presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -112,6 +122,246 @@ When a ``GROUP BY`` clause is used in a ``SELECT`` statement all output
 expression must be either aggregate functions or columns present in
 the ``GROUP BY`` clause.
 
+**Complex Grouping Operations**
+
+Presto also supports complex aggregations using the ``GROUPING SETS``, ``CUBE``
+and ``ROLLUP`` syntax. This syntax allows users to perform analysis that require
+aggregation on multiple sets of columns in a single query. Complex grouping
+operations do not support grouping on expressions composed of input columns.
+Only column names or ordinals are allowed.
```

Comment:
Even though the generated plans are identical, I usually prefer this query 
`select cast(event_time as date), count(*) from table group by 1` over
`select cast(event_time as date), count(*) from table group by cast(event_time as date)`.
Later on, I got this habit of using column ordinals in GROUP BY but I didn't know that it's anti-pattern. Thanks @martint!
BTW, I think we should update the docs if referencing column ordinals is not allowed in GROUPING SETS, CUBE, ROLLUP.